### ISSUE TITLE
Create Sequence Diagram of existing CSV driver

### ISSUE THREAD

**Issue by SpencerFleming** [BoiseState-AdaptLab/adapt-lidar-tools#272]
I want to create some UML diagrams for the program this time to allow me to implement the program with a more regimented approach (which, knowing me, should play on my strengths as a top-down thinker).

EDIT: I discovered that a sequence diagram actually does a much better job at conveying the information I need in order to start designing a better program. I may have other types of diagrams to accompany it though, just to give it context, but the already-existing UML diagrams can do most of that work for me.

**Acceptance Criteria**
- [x] A link to a sequence diagram that shows how CSV driver works in its current state.

**Other notes**
Once this is done I'll use it to plan the new features we need in a separate task.

**Comment by SpencerFleming** [comment]
In doing this, I identified a general four-step process that the software undergoes to create the final product.
1. Configurations are parsed and stored in CmdLine (CmdLine).
2. Processor objects are set up and configured per CmdLine's specifications (main/Lidar Driver).
2. Processing is executed (Gaussian Fitter).
3. Products are generated (Lidar Driver / Csv Writer / etc)

Thus, our process lends itself to keeping these steps organized separately for simplicity and ease of use (notice how Lidar Driver is involved in two steps, violating this concept).

The two configurable parts being added to pls_to_csv:
1) Input can be from pls OR from txt
2) Outputs can be diagnostic information from gaussian fitting in txt file AND/OR csv file of peaks

Step 1 will give us the information on which of these are being used. Step 2 will make sure it happens by setting up the right objects, and configuring objects in the right way. Then step 2 and 3 will automatically do what we need. That's the idea.

In conclusion, when I design the classes and functions included in this Epic, they should be able to be configured completely prior to the processing occurring, so when processing occurs, it can be an easy to code, easy to understand, and fast process.
